bofire/data_models/constraints/multilinear.py

```diff
@@ -0,0 +1,106 @@
+from typing import Annotated, List, Literal
+
+import numpy as np
+import pandas as pd
+from pydantic import Field, model_validator
+
+from bofire.data_models.constraints.constraint import (
+    EqalityConstraint,
+    InequalityConstraint,
+    IntrapointConstraint,
+)
+from bofire.data_models.types import FeatureKeys
+
+
+class MultiLinearConstraint(IntrapointConstraint):
+    """
+    Represents a multi-linear constraint of the form `sign * x1**e1 * x2**e2 * ... * xn**en`.
+
+    Attributes:
+        type (str): The type of the constraint.
+        features (FeatureKeys): The keys of the features used in the constraint.
+        exponents (List[float]): The exponents corresponding to each feature.
```

Comment:
Exponents are of type float but I guess you assume to only receive integers because otherwise it wouldn't be multi-linear anymore, right? Can you do `List[int]` as type hint? Or document it at least somehow?

Reply:
Hmm, it also works with floats, so I would like to keep it as `List[float]`, do you have a better name for it?

Reply:
Hmmm... it's a polynomial with real exponants, right? I just had a look. There seems to be a name for that but I have not heard it before.
https://en.wikipedia.org/wiki/Posynomial

Reply:
Yep, only product, for me it is somehow a generalized multilinear one ..., so is this a problem for you to keep it?

Reply:
Giving it another thought, I would simply call it the `ProductConstraint`. I think, multi-linear means it is linear in each variable which this is something different.

Reply:
Done.